### ISSUE TITLE
Persistent Volume Claims of /downloads and /movies should be located on same volume with one pvc

### ISSUE THREAD

**Issue by abel408** [billimek/billimek-charts#222]
By default, radarr and sonarr are set to rename movies and create hardlinks. Hardlinks can only be created when both file locations are located on the same file system. Even though the /downloads and /movies PV could be from the same share, the container has two mount points and will not recognize that they are from the same system. Instead of creating hardlinks, radarr and sonarr will copy the file.

**Comment by billimek** [comment]
Hi @abel408, correct!  We have the [`extraExistingClaimMounts`](https://github.com/billimek/billimek-charts/blob/master/charts/sonarr/values.yaml#L142-L148) configuration option to help address this (also for radarr)

An example of the implementation is [here](https://github.com/billimek/k8s-gitops/blob/master/default/sonarr/sonarr.yaml#L30-L43):

```yaml
    persistence:
      config:
        enabled: true
        storageClass: "rook-ceph-block"
        size: 2Gi
      downloads:
        enabled: false
      tv:
        enabled: false
      extraExistingClaimMounts:
        - name: media
          existingClaim: nfs-media-pvc
          mountPath: /media
          readOnly: false
```

It seems like we should still 'allow' someone to choose to have the different mounts for downloads and tv/movies.  What do you think?

**Comment by abel408** [comment]
Ahhh, thank you. I do agree with allowing different mount points, however, the chart by default will create a mount point for each and radarr and sonarr, by default, will use hardlinks so I think there is a bit of a discrepancy there. I would imagine the only reason someone would want a mount point for each would be if they are organizing things manually? I just learned the hard way that radarr and sonarr have been duplicating all my files and am now in the process of reorganizing :/. I have a feeling I'm not the only one.

**Comment by billimek** [comment]
@abel408 this is a good point, we should consider changing the default behavior.  Something different than the `extraExistingClaimMounts` would be necessary for the two charts because we won't have an existing claim to reference by default.

@onedr0p what do you think?  I think we would need to change the chart to collapse /downloads and /tv or /movies into a single PVC and allow folks to leverage extraExistingClaimMounts if they really want them separate.  We could bump the chart major version to indicate a breaking change.

**Comment by onedr0p** [comment]
@billimek do you think a single `media` volume is good instead of `downloads` and `tv` or `movies` ?

**Comment by billimek** [comment]
@onedr0p yeah, exactly!  Can you think of any downsides taking that approach?

**Comment by onedr0p** [comment]
@billimek maybe, my current deployments use media in existingVolumeClaims. Do you think that is a conflict?

```yml
    persistence:
      downloads:
        enabled: false
      tv:
        enabled: false
      extraExistingClaimMounts:
        - name: media
          existingClaim: nfs-media-pvc
          mountPath: /media
          readOnly: false
```

**Comment by billimek** [comment]
With the change we're discussing, yeah.  I see it as being something like this:

```yaml
    persistence:
      config:
        enabled: true
      media:
        enabled: true
        existingClaim: nfs-media-pvc
        subPath: /media
      extraExistingClaimMounts: <not needed now>
```

... which would be a 'breaking' change

**Comment by onedr0p** [comment]
Sounds good! I think this would be a great change.

**Comment by WojoInc** [comment]
Took a stab at this, as I actually just ran into the same issue, albeit on plain Docker. While bazarr doesn't necessarily have the same issue with linking, figured all the 'arr apps should all probably be laid out the same.
If the commits [here](https://github.com/WojoInc/billimek-charts/tree/merge_pvc) look good/cohesive, I can go ahead and do the same for sonarr, etc. and toss a PR together.

**Comment by onedr0p** [comment]
@WojoInc looks great, feel free to open PRs :)

**Comment by WojoInc** [comment]
Sg! Submitted PR for Bazarr. I plan on doing the same for Sonarr, Radarr, and Lidarr after work today. Any other charts that may need the same thing done?

**Comment by billimek** [comment]
Should be fixed now by #224, #225, #226, & #227

**Comment by windowsrefund** [comment]
As someone who is starting out with these charts, I'm not worried about backwards compatibility or breaking changes and just want to get things up and running as cleanly as possible. Is there some documentation or walk-through published anywhere that would show how to create a sane values.yaml (to be passed into the install via -f) which results in sonarr using the same volume for radarr and plex? In other words, all 3 would end up using the same PVC?

**Comment by onedr0p** [comment]
@windowsrefund what you want is NFS storage but radarr and Sonarr don't support it, it's better to use block volumes

**Comment by windowsrefund** [comment]
Hi, I'm actually using cStor from OpenEBS

```
kubectl -n media get pvc
NAME            STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS            AGE
sonarr-config   Bound    pvc-b3597064-7d8f-48c7-944d-0841d4814868   1Gi        RWO            openebs-cstor-persist   23m
sonarr-media    Bound    pvc-587c130d-5362-4c2a-ac8f-1fcafcaf70fa   10Gi       RWO            openebs-cstor-persist   23m
```
I've sucessfully installed this chart as follows:
`helm -n media install sonarr billimek/sonarr -f charts/sonarr/values.yaml`

Where my values.yaml specifies the PVCs and makes use of  `skipuninstall: true`. Now that I've verified Sonar sees /config and /media, I'm thinking I did this correctly. Soon, I'll go ahead and update my values.yaml in order to grab all the disk I can for the sonarr-media PVC.

The question I'm asking is more related to when I add additional applications to my cluster like plex, nzbget, transmission, and radarr. I'd really want them to all use the same storage (as is used by the sonarr-media PVC). Does that mean I'd want to tell each of those apps to use that exact PVC?

Just trying to get my head around how to properly ties all these together from a storage perspective. Also, sorry if I hijacked this thread. I'd be more than happy to continue the conversation elsewhere if that would be appropriate.
